### PR TITLE
HPCC-13029 Clean up unit tests

### PR DESCRIPTION
**Pull request by richardkchapman**
Rework the unittest program to allow finer control over what tests are to be
run.

Delay initialization on Dali unit tests until it is proven they are included.

Signed-off-by: Richard Chapman rchapman@hpccsystems.com

### CODE REVIEW

File: testing/unittests/unittests.cpp [hunk 1]
```diff
@@ -37,34 +39,201 @@
...
 
+void usage()
+{
+    printf("\n"
+    "Usage:\n"
+    "    unittests <options> <testnames>\n"
+    "\n"
+    "Options:\n"
+    "    -h  --help     Display this help text\n"
+    "    -l  --list     List matching tests but do not execute them\n"
```
Comment (mckellyln) [inline testing/unittests/unittests.cpp]: Does -l list all available tests ?  If so, a welcome addition.

Comment (richardkchapman) [inline testing/unittests/unittests.cpp]: Yes

**Comment by richardkchapman** [comment]
@ghalliday Please review

File: roxie/roxiemem/roxiemem.cpp [hunk 2]
```diff
@@ -5542,18 +5542,14 @@ class RoxieMemTests : public CppUnit::TestFixture
...
-        CPPUNIT_TEST(testCas);
         CPPUNIT_TEST(testAll);
-        CPPUNIT_TEST(testCallbacks);
         CPPUNIT_TEST(testRecursiveCallbacks);
-        CPPUNIT_TEST(testCompacting);
         CPPUNIT_TEST(testResize);
         //MORE: The following currently leak pages, so should go last
         CPPUNIT_TEST(testDatamanager);
-        CPPUNIT_TEST(testDatamanagerThreading);
+//        CPPUNIT_TEST(testDatamanagerThreading);  stress
```
Comment (ghalliday) [inline roxie/roxiemem/roxiemem.cpp]: minor: Delete this comment

**Pull request merged**